### ISSUE TITLE
There are compatibility issues between Create, EMI, and Farmer's Delight

### ISSUE THREAD

**Issue by Chikage0o0** [newhoryzon/farmers-delight-fabric#115]
Adding Farmer's Delight when both Create and EMI are loaded will cause EMI to fail to load some of Create's recipes

	- create 0.5.0g-796+1.19.2
	- farmersdelight 1.19.2-1.3.9
	- emi 0.5.0+1.19.2
	- fabric-api 0.68.0+1.19.2

ERROR：
```
[19:53:41] [Thread-12/ERROR]: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base
[19:53:41] [Thread-12/ERROR]: jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base
[19:53:41] [Thread-12/ERROR]: jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base
[19:53:41] [Thread-12/ERROR]: jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base
[19:53:41] [Thread-12/ERROR]: java.util.Objects.checkIndex(Objects.java:359)
	at java.base
[19:53:41] [Thread-12/ERROR]: java.util.ArrayList.get(ArrayList.java:427)
	at com.simibubi.create.compat.emi.recipes.SpoutEmiRecipe.<init>(SpoutEmiRecipe.java:19)
	at com.simibubi.create.compat.emi.CreateEmiPlugin.addAll(CreateEmiPlugin.java:283)
	at com.simibubi.create.compat.emi.CreateEmiPlugin.register(CreateEmiPlugin.java:269)
	at dev.emi.emi.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:112)
	at java.base
[19:53:41] [Thread-12/ERROR]: java.lang.Thread.run(Thread.java:833)
```

**Comment by JieningYu** [comment]
dupe, cringe

**Comment by JieningYu** [comment]
please provide the full log

**Comment by Chikage0o0** [comment]
> please provide the full log

```
[19:51:32] [main/INFO]: Loading Minecraft 1.19.2 with Fabric Loader 0.14.11
[19:51:32] [ForkJoinPool-1-worker-2/WARN]: Mod create uses the version 0.5.0g-796+1.19.2 which isn't compatible with Loader's extended semantic version format (Could not parse version number component '0g'!), SemVer is recommended for reliably evaluating dependencies and prioritizing newer version
[19:51:32] [main/INFO]: Loading 86 mods:
	- com_electronwill_night-config_core 3.6.5 via forgeconfigapiport
	- com_electronwill_night-config_toml 3.6.5 via forgeconfigapiport
	- com_github_llamalad7_mixinextras 0.1.1 via porting_lib_base
	- com_google_code_findbugs_jsr305 3.0.2 via create
	- create 0.5.0g-796+1.19.2
	- dripstone_fluid_lib 2.0.1 via milk
	- emi 0.5.0+1.19.2
	- fabric-api 0.68.0+1.19.2
	- fabric-api-base 0.4.13+0d0f210290 via fabric-api
	- fabric-api-lookup-api-v1 1.6.12+93d8cb8290 via fabric-api
	- fabric-biome-api-v1 9.0.20+0d0f210290 via fabric-api
	- fabric-block-api-v1 1.0.1+12bfe4ea90 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.22+c6af733c90 via fabric-api
	- fabric-client-tags-api-v1 1.0.3+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.13+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.1.9+93d8cb8290 via fabric-api
	- fabric-commands-v0 0.2.30+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.38+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.4.3+6064db2a90 via fabric-api
	- fabric-convention-tags-v1 1.1.3+93d8cb8290 via fabric-api
	- fabric-crash-report-info-v1 0.2.7+aeb40ebe90 via fabric-api
	- fabric-data-generation-api-v1 5.3.2+ad31814d90 via fabric-api
	- fabric-dimensions-v1 2.1.33+0d0f210290 via fabric-api
	- fabric-entity-events-v1 1.5.2+9244241690 via fabric-api
	- fabric-events-interaction-v0 0.4.31+6eb9a2a190 via fabric-api
	- fabric-events-lifecycle-v0 0.2.31+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.23+c6af733c90 via fabric-api
	- fabric-item-api-v1 1.6.1+fa140d5990 via fabric-api
	- fabric-item-groups-v0 0.3.35+9244241690 via fabric-api
	- fabric-key-binding-api-v1 1.0.23+aaaf9d3390 via fabric-api
	- fabric-keybindings-v0 0.2.21+df3654b390 via fabric-api
	- fabric-lifecycle-events-v1 2.2.2+9244241690 via fabric-api
	- fabric-loot-api-v2 1.1.9+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.12+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.5+93d8cb8290 via fabric-api
	- fabric-mining-level-api-v1 2.1.20+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.19+c6af733c90 via fabric-api
	- fabric-networking-api-v1 1.2.8+3b1d070f90 via fabric-api
	- fabric-networking-v0 0.3.25+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.1.2+06e0e25190 via fabric-api
	- fabric-particles-v1 1.0.12+79adfe0a90 via fabric-api
	- fabric-registry-sync-v0 0.9.29+9244241690 via fabric-api
	- fabric-renderer-api-v1 1.1.0+e7aeff4590 via fabric-api
	- fabric-renderer-indigo 0.6.17+aeb40ebe90 via fabric-api
	- fabric-renderer-registries-v1 3.2.22+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.17+0d0f210290 via fabric-api
	- fabric-rendering-fluids-v1 3.0.9+c6af733c90 via fabric-api
	- fabric-rendering-v0 1.1.25+df3654b390 via fabric-api
	- fabric-rendering-v1 1.11.1+73145abb90 via fabric-api
	- fabric-resource-conditions-api-v1 2.1.1+aae9039d90 via fabric-api
	- fabric-resource-loader-v0 0.8.1+0d0f210290 via fabric-api
	- fabric-screen-api-v1 1.0.29+0d0f210290 via fabric-api
	- fabric-screen-handler-api-v1 1.3.4+1cc24b1b90 via fabric-api
	- fabric-sound-api-v1 1.0.1+c4f28df590 via fabric-api
	- fabric-textures-v0 1.0.22+aeb40ebe90 via fabric-api
	- fabric-transfer-api-v1 2.1.3+9d081d6490 via fabric-api
	- fabric-transitive-access-wideners-v1 1.3.2+42d99c3290 via fabric-api
	- fabricloader 0.14.11
	- fake-player-api 0.5.0 via create
	- farmersdelight 1.19.2-1.3.9
	- flywheel 0.6.8-1 via create
	- forge_tags 3.0 via porting_lib
	- forgeconfigapiport 4.2.6 via create
	- java 17
	- milk 1.0.51 via create
	- minecraft 1.19.2
	- mm 2.3 via porting_lib_base
	- noindium 1.1.0+1.19 via create
	- porting_lib 2.1.623+1.19.2 via create
	- porting_lib_accessors 2.1.623+1.19.2 via porting_lib_extensions
	- porting_lib_attributes 2.1.623+1.19.2 via porting_lib_extensions
	- porting_lib_base 2.1.623+1.19.2 via porting_lib
	- porting_lib_common 2.1.623+1.19.2 via porting_lib_extensions
	- porting_lib_constants 2.1.623+1.19.2 via porting_lib_attributes
	- porting_lib_entity 2.1.623+1.19.2 via porting_lib
	- porting_lib_extensions 2.1.623+1.19.2 via porting_lib_transfer
	- porting_lib_lazy_registration 2.1.623+1.19.2 via porting_lib
	- porting_lib_model_generators 2.1.623+1.19.2 via porting_lib
	- porting_lib_model_loader 2.1.623+1.19.2 via porting_lib_obj_loader
	- porting_lib_models 2.1.623+1.19.2 via porting_lib
	- porting_lib_networking 2.1.623+1.19.2 via porting_lib
	- porting_lib_obj_loader 2.1.623+1.19.2 via porting_lib
	- porting_lib_transfer 2.1.623+1.19.2 via porting_lib
	- reach-entity-attributes 2.3.0 via create
	- registrate-fabric 1.1.44-MC1.19.2 via create
	- serialization_hooks 0.3.26 via porting_lib_base
[19:51:35] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/D:/Games/mc/.minecraft/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[19:51:36] [main/INFO]: Compatibility level set to JAVA_17
[19:51:36] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/chunk/data/ChunkRenderData$Builder (java.lang.ClassNotFoundException: me/jellysquid/mods/sodium/client/render/chunk/data/ChunkRenderData$Builder)
[19:51:36] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.chunk.data.ChunkRenderData$Builder was not found flywheel.mixins.json:instancemanage.SodiumChunkRenderDataMixin from mod flywheel
[19:51:37] [main/WARN]: Configuration conflict: there is more than one oshi.properties file on the classpath
[19:51:38] [main/WARN]: Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[19:51:40] [main/WARN]: Method overwrite conflict for setTranslation in flywheel.fabric.mixins.json:Matrix4fMixin from mod flywheel, previously written by io.github.fabricators_of_create.porting_lib.mixin.common.Matrix4fMixin. Skipping method.
[19:51:41] [main/INFO]: Building optimized datafixer
[19:51:48] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[19:51:50] [Render thread/INFO]: Setting user: Chikage0v0
[19:51:55] [Render thread/WARN]: Encountered duplicate API provider registration for item: minecraft:glass_bottle
[19:51:55] [Render thread/WARN]: Encountered duplicate API provider registration for item: minecraft:bucket
[19:51:55] [Render thread/WARN]: Encountered duplicate API provider registration for item: minecraft:glass_bottle
[19:51:55] [Render thread/WARN]: Encountered duplicate API provider registration for item: minecraft:bucket
[19:51:57] [Render thread/WARN]: Encountered duplicate API provider registration for item: minecraft:bucket
[19:51:58] [Render thread/INFO]: Definitely not up to no good
[19:51:59] [Render thread/INFO]: [Indigo] Registering Indigo renderer!
[19:52:00] [Render thread/INFO]: No shaders mod detected.
[19:52:00] [Render thread/INFO]: Trying to read config file...
[19:52:00] [Render thread/INFO]: A config file was found, loading it..
[19:52:00] [Render thread/INFO]: Successfully loaded config file.
[19:52:01] [Render thread/INFO]: Backend library: LWJGL version 3.3.1 SNAPSHOT
[19:52:03] [Render thread/INFO]: Narrator library for x64 successfully loaded
[19:52:03] [Render thread/INFO]: Reloading ResourceManager: Default, Fabric Mods (Create, Dripstone Fluid Lib, EMI, Fabric API, Fabric API Base, Fabric API Lookup API (v1), Fabric Biome API (v1), Fabric BlockRenderLayer Registration (v1), Fabric Client Tags, Fabric Command API (v1), Fabric Command API (v2), Fabric Commands (v0), Fabric Containers (v0), Fabric Content Registries (v0), Fabric Convention Tags, Fabric Crash Report Info (v1), Fabric Data Generation API (v1), Fabric Dimensions API (v1), Fabric Entity Events (v1), Fabric Events Interaction (v0), Fabric Events Lifecycle (v0), Fabric Game Rule API (v1), Fabric Item API (v1), Fabric Item Groups (v0), Fabric Key Binding API (v1), Fabric Key Bindings (v0), Fabric Lifecycle Events (v1), Fabric Loot API (v2), Fabric Loot Tables (v1), Fabric Message API (v1), Fabric Mining Level API (v1), Fabric Models (v0), Fabric Networking API (v1), Fabric Networking (v0), Fabric Object Builder API (v1), Fabric Particles (v1), Fabric Registry Sync (v0), Fabric Renderer API (v1), Fabric Renderer - Indigo, Fabric Renderer Registries (v1), Fabric Rendering Data Attachment (v1), Fabric Rendering Fluids (v1), Fabric Rendering (v0), Fabric Rendering (v1), Fabric Resource Conditions API (v1), Fabric Resource Loader (v0), Fabric Screen API (v1), Fabric Screen Handler API (v1), Fabric Sound API (v1), Fabric Textures (v0), Fabric Transfer API (v1), Fabric Transitive Access Wideners (v1), Fabric Loader, Fake Player API, Farmer's Delight, Flywheel, Porting Lib Tags, Forge Config API Port, Milk Lib, Manningham Mills, No Indium?, Porting Lib, Porting Lib Accessors, Porting Lib Attributes, Porting Lib Base, Porting Lib Common, Porting Lib Constants, Porting Lib Entity, Porting Lib Extensions, Porting Lib Lazy Register, Porting Lib Model Generators, Porting Lib Model Loader, Porting Lib Models, Porting Lib Networking, Porting Lib Obj Loader, Porting Lib Transfer, Reach Entity Attributes, Registrate for Fabric, Serialization Hooks)
[19:52:05] [Worker-Bootstrap-6/INFO]: 4283 Datafixer optimizations took 24087 milliseconds
[19:52:07] [Render thread/WARN]: Missing sound for event: minecraft:item.goat_horn.play
[19:52:07] [Render thread/WARN]: Missing sound for event: minecraft:entity.goat.screaming.horn_break
[19:52:07] [Render thread/INFO]: OpenAL initialized on device OpenAL Soft on 扬声器 (13- DA800)
[19:52:07] [Render thread/INFO]: Sound engine started
[19:52:08] [Render thread/INFO]: Created: 2048x2048x4 minecraft:textures/atlas/blocks.png-atlas
[19:52:08] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/signs.png-atlas
[19:52:08] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/banner_patterns.png-atlas
[19:52:08] [Render thread/INFO]: Created: 1024x512x4 minecraft:textures/atlas/shield_patterns.png-atlas
[19:52:08] [Render thread/INFO]: Created: 256x256x4 minecraft:textures/atlas/chest.png-atlas
[19:52:08] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/beds.png-atlas
[19:52:08] [Render thread/INFO]: Created: 512x256x4 minecraft:textures/atlas/shulker_boxes.png-atlas
[19:52:09] [Render thread/WARN]: Shader rendertype_entity_translucent_emissive could not find sampler named Sampler2 in the specified shader program.
[19:52:09] [Render thread/INFO]: Created: 512x512x0 minecraft:textures/atlas/particles.png-atlas
[19:52:09] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/paintings.png-atlas
[19:52:09] [Render thread/INFO]: Created: 256x128x0 minecraft:textures/atlas/mob_effects.png-atlas
[19:52:09] [Render thread/INFO]: Loaded all shader sources.
[19:53:31] [Render thread/ERROR]: IngredientDeserializer with ID not found: [farmersdelight:tool] this can be ignored unless issues occur.
[19:53:31] [Render thread/WARN]: Suspicious fluid amount in recipe [farmersdelight:create/filling/chocolate_pie]: 500
[19:53:31] [Render thread/WARN]: Suspicious fluid amount in recipe [farmersdelight:create/mixing/pie_crust_from_mixing]: 250
[19:53:31] [Render thread/INFO]: Loaded 24 recipes
[19:53:32] [Render thread/INFO]: Loaded 2566 advancements
[19:53:34] [Render thread/INFO]: Applied 199 biome modifications to 63 of 63 new biomes in 14.53 ms
[19:53:34] [Server thread/INFO]: Registering compost heaps in village type of plains
[19:53:34] [Server thread/INFO]: Registering compost heaps in village type of savanna
[19:53:34] [Server thread/INFO]: Registering compost heaps in village type of snowy
[19:53:34] [Server thread/INFO]: Registering compost heaps in village type of taiga
[19:53:34] [Server thread/INFO]: Registering compost heaps in village type of desert
[19:53:34] [Server thread/INFO]: Starting integrated minecraft server version 1.19.2
[19:53:34] [Server thread/INFO]: Generating keypair
[19:53:34] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[19:53:36] [Render thread/INFO]: 准备生成区域中：0%
[19:53:36] [Render thread/INFO]: 准备生成区域中：0%
[19:53:36] [Render thread/INFO]: 准备生成区域中：0%
[19:53:36] [Render thread/INFO]: 准备生成区域中：0%
[19:53:36] [Render thread/INFO]: 准备生成区域中：0%
[19:53:37] [Render thread/INFO]: 准备生成区域中：4%
[19:53:37] [Render thread/INFO]: Time elapsed: 3034 ms
[19:53:37] [Server thread/INFO]: Changing view distance to 12, from 10
[19:53:37] [Server thread/INFO]: Changing simulation distance to 12, from 0
[19:53:39] [Netty Local Client IO #0/INFO]: Connected to a modded server.
[19:53:39] [Server thread/INFO]: Chikage0v0[local:E:246e2c66] logged in with entity id 162 at (181.14660513870302, 72.0, -134.00454568502565)
[19:53:39] [Server thread/INFO]: Chikage0v0加入了游戏
[19:53:39] [Render thread/INFO]: Started 10 worker threads
[19:53:40] [Thread-12/INFO]: Starting EMI reload...
[19:53:40] [Render thread/INFO]: Loaded 0 advancements
[19:53:40] [Thread-12/INFO]: Reloaded plugin from EMI in 334ms
[19:53:41] [Thread-12/ERROR]: [emi] Exception loading plugin provided by create
[19:53:41] [Thread-12/INFO]: [STDERR]: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at java.base/java.util.Objects.checkIndex(Objects.java:359)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at java.base/java.util.ArrayList.get(ArrayList.java:427)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at com.simibubi.create.compat.emi.recipes.SpoutEmiRecipe.<init>(SpoutEmiRecipe.java:19)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at com.simibubi.create.compat.emi.CreateEmiPlugin.addAll(CreateEmiPlugin.java:283)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at com.simibubi.create.compat.emi.CreateEmiPlugin.register(CreateEmiPlugin.java:269)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at dev.emi.emi.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:112)
[19:53:41] [Thread-12/INFO]: [STDERR]: 	at java.base/java.lang.Thread.run(Thread.java:833)
[19:53:41] [Thread-12/ERROR]: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base
[19:53:41] [Thread-12/ERROR]: jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base
[19:53:41] [Thread-12/ERROR]: jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base
[19:53:41] [Thread-12/ERROR]: jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base
[19:53:41] [Thread-12/ERROR]: java.util.Objects.checkIndex(Objects.java:359)
	at java.base
[19:53:41] [Thread-12/ERROR]: java.util.ArrayList.get(ArrayList.java:427)
	at com.simibubi.create.compat.emi.recipes.SpoutEmiRecipe.<init>(SpoutEmiRecipe.java:19)
	at com.simibubi.create.compat.emi.CreateEmiPlugin.addAll(CreateEmiPlugin.java:283)
	at com.simibubi.create.compat.emi.CreateEmiPlugin.register(CreateEmiPlugin.java:269)
	at dev.emi.emi.EmiReloadManager$ReloadWorker.run(EmiReloadManager.java:112)
	at java.base
[19:53:41] [Thread-12/ERROR]: java.lang.Thread.run(Thread.java:833)

[19:53:41] [Thread-12/INFO]: Baked 6440 recipes in 353ms
[19:53:41] [Thread-12/INFO]: Reloaded EMI in 1080ms
[19:53:42] [Server thread/INFO]: Saving and pausing game...
[19:53:42] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:overworld
[19:53:42] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_nether
[19:53:42] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_end
[19:53:51] [Server thread/INFO]: Saving and pausing game...
[19:53:51] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:overworld
[19:53:51] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_nether
[19:53:51] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_end
[19:53:52] [Server thread/INFO]: Saving and pausing game...
[19:53:52] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:overworld
[19:53:52] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_nether
[19:53:52] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_end
[19:54:16] [Server thread/INFO]: Saving and pausing game...
[19:54:16] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:overworld
[19:54:16] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_nether
[19:54:16] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_end
[20:00:57] [Server thread/INFO]: Saving and pausing game...
[20:00:57] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:overworld
[20:00:57] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_nether
[20:00:57] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_end
[20:02:57] [Render thread/INFO]: Stopping!
[20:02:57] [Server thread/INFO]: Chikage0v0 lost connection: 连接中断
[20:02:57] [Server thread/INFO]: Chikage0v0退出了游戏
[20:02:57] [Server thread/INFO]: Stopping singleplayer server as player logged out
[20:02:57] [Server thread/INFO]: Stopping server
[20:02:57] [Server thread/INFO]: Saving players
[20:02:57] [Server thread/INFO]: Saving worlds
[20:02:58] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:overworld
[20:02:58] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_nether
[20:02:58] [Server thread/INFO]: Saving chunks for level 'ServerLevel[新的世界1]'/minecraft:the_end
[20:02:58] [Server thread/INFO]: ThreadedAnvilChunkStorage (新的世界1): All chunks are saved
[20:02:58] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[20:02:58] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[20:02:58] [Server thread/INFO]: ThreadedAnvilChunkStorage: All dimensions are saved

```

**Comment by JieningYu** [comment]
this should not caused by farmer's delight.

**Comment by Chikage0o0** [comment]
> this should not caused by farmer's delight.
No farmers' joy, no problems

**Comment by DoggoNako** [comment]
Has he had any research in the last 3 days? I also have this problem that blocks me from seeing all the Create recipes using EMI. *You can see the mention above with the logs*

**Comment by LemmaEOF** [comment]
I found the bug! It's the Farmer's Delight Create-compat [chocolate pie filling recipe](https://github.com/newhoryzon/farmers-delight-fabric/blob/master/src/main/resources/data/farmersdelight/recipes/create/filling/chocolate_pie.json) - the fluid tag `c:chocolates` isn't used in Create, it's `c:chocolate` instead, so switching to the proper tag should stop it from getting index out-of-bounds issues!

**Comment by MrBrax** [comment]
Is this getting a fix? it's happening with JEI too. I have no idea if i can fix it myself.

**Comment by DoggoNako** [comment]
https://github.com/newhoryzon/farmers-delight-fabric/pull/116 It's been 1 month since the last update, idk when the update will be released.

**Comment by Abalieno** [comment]
This is easily fixable modpack-side through kubejs (or jar editing, I guess).

Just add the correct recipe to kubejs\data\farmersdelight\recipes\create\filling\chocolate_pie.json

**Comment by Zifiv** [comment]
Should be fixed by #116.
